pkg/miniogw/gateway-storj.go

```diff
@@ -290,18 +290,18 @@ func (s *storjObjects) ListObjectsV2(ctx context.Context, bucket, prefix, contin
 
 	if len(items) > 0 {
 		for _, item := range items {
-			p := item.Path
-			if recursive {
-				p = path.Join(prefix, p)
+			path := item.Path
+			if recursive && prefix != "" {
+				path = storj.JoinPaths(strings.TrimSuffix(prefix, "/"), path)
```

Comment:
`prefix + path` should work directly here otherwise it means somewhere the splitting is problematic.